src/main/java/org/reaktivity/nukleus/http_cache/internal/proxy/cache/DefaultCache.java

```diff
@@ -372,14 +402,27 @@ public boolean isRequestCacheable(
     public void purgeEntriesForNonPendingRequests(
         Set<Integer> requestHashes)
     {
-        cachedEntriesByRequestHash.forEach((requestHash, cacheEntry) ->
+        for (final MutableInteger count = new MutableInteger(1); count.value <= allowedCacheEvictionCount;)
         {
-            if (!requestHashes.contains(requestHash))
+            final FrequencyBucket frequencyBucket = frequencies.get(count.value);
+            frequencyBucket.entries().forEach(entry ->
             {
-                purge(requestHash);
-            }
-        });
-
+                final int requestHash = entry.requestHash();
+                if (!requestHashes.contains(requestHash))
+                {
+                    if (count.value <= allowedCacheEvictionCount)
+                    {
+                        purge(requestHash);
+                        count.value++;
+                    }
+                    else
+                    {
+                        return;
+                    }
+                }
+            });
+        }
+        counters.cacheFullness.getAsLong();
```

Comment:
Suggest renaming to something like `cachePurgeAttempts` since it seems to be tracking the number of times we attempt to purge.